kotlin/codegen/src/main/java/com/squareup/moshi/kotlin/codegen/TargetType.kt

```diff
@@ -204,13 +204,19 @@ internal data class TargetType(
 
     private fun genericTypeNames(proto: Class, nameResolver: NameResolver): List<TypeVariableName> {
       return proto.typeParameterList.map {
+        val possibleBounds = it.upperBoundList
```

Comment:
Yeah, this is something we can improve. Right now, it's safe to call `withBounds(emptyList())` on a `TypeVariableName`, as it will retain the implicit `Any?` bound.

Reply:
https://github.com/square/kotlinpoet/issues/420